crates/pallet-executor/src/lib.rs

```diff
@@ -308,6 +324,29 @@ mod pallet {
 }
 
 impl<T: Config> Pallet<T> {
+    fn check_execution_receipt(
+        SignedExecutionReceipt {
+            execution_receipt,
+            signature,
+            signer,
+        }: &SignedExecutionReceipt<T::SecondaryHash>,
+    ) -> Result<(), Error<T>> {
+        let msg = execution_receipt.hash().encode();
+        if !signer.verify(&msg, signature) {
+            return Err(Error::<T>::BadExecutionReceiptSignature);
+        }
+
+        // TODO: upgrade once the trusted executor system is upgraded.
+        let expected_executor = Self::executor()
```

Comment:
lets fail early even before we check for signature validity ?

Reply:
Hmm, why?

Reply:
Probably because this check is cheaper than verifying signature. Though it is not supposed to ever crash anyway, so shouldn't matter.

Reply:
This check needs to read the state while verifying the signature only consumes CPU and the unsigned extrinsic validation is invoked by the transaction pool, so I'm not fairly sure it's more beneficial.